azurerm/helpers/validate/servicebus_namespace.go

```diff
@@ -1,164 +1,42 @@
-	)
-}
-
-func ValidateServiceBusSubscriptionName() schema.SchemaValidateFunc {
-	return validation.StringMatch(
-		regexp.MustCompile("^[a-zA-Z0-9][-._a-zA-Z0-9]{0,48}([a-zA-Z0-9])?$"),
-		"The name can contain only letters, numbers, periods, hyphens and underscores. The name must start and end with a letter or number and be less then 50 characters long.",
-	)
-}
-
-func ValidateServiceBusTopicName() schema.SchemaValidateFunc {
-	return validation.StringMatch(
-		regexp.MustCompile("^[a-zA-Z][-._~a-zA-Z0-9]{0,258}([a-zA-Z0-9])?$"),
-		"The topic name can contain only letters, numbers, periods, hyphens, tildas and underscores. The namespace must start with a letter, and it must end with a letter or number and be less then 260 characters long.",
-	)
-}
-
-func ValidateServiceBusAuthorizationRuleName() schema.SchemaValidateFunc {
-	return validation.StringMatch(
-		regexp.MustCompile("^[a-zA-Z0-9][-._a-zA-Z0-9]{0,48}([a-zA-Z0-9])?$"),
-		"The name can contain only letters, numbers, periods, hyphens and underscores. The name must start and end with a letter or number and be less the 50 characters long.",
-	)
-}
-
-func ExpandServiceBusAuthorizationRuleRights(d *schema.ResourceData) *[]servicebus.AccessRights {
-	rights := make([]servicebus.AccessRights, 0)
-
-	if d.Get("listen").(bool) {
-		rights = append(rights, servicebus.Listen)
-	}
-
-	if d.Get("send").(bool) {
-		rights = append(rights, servicebus.Send)
-	}
-
-	if d.Get("manage").(bool) {
-		rights = append(rights, servicebus.Manage)
-	}
-
-	return &rights
-}
-
-func FlattenServiceBusAuthorizationRuleRights(rights *[]servicebus.AccessRights) (listen, send, manage bool) {
-	// zero (initial) value for a bool in go is false
-
-	if rights != nil {
-		for _, right := range *rights {
-			switch right {
-			case servicebus.Listen:
-				listen = true
-			case servicebus.Send:
-				send = true
-			case servicebus.Manage:
-				manage = true
-			default:
-				log.Printf("[DEBUG] Unknown Authorization Rule Right '%s'", right)
-			}
-		}
-	}
-
-	return listen, send, manage
-}
-
-// shared schema
-func MergeSchema(a map[string]*schema.Schema, b map[string]*schema.Schema) map[string]*schema.Schema {
-	s := map[string]*schema.Schema{}
-
-	for k, v := range a {
-		s[k] = v
-	}
+func ServiceBusNamespaceName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
 
-	for k, v := range b {
-		s[k] = v
+	if !regexp.MustCompile("[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}").MatchString(value) || !regexp.MustCompile("^[a-zA-Z][-a-zA-Z0-9]{4,48}[a-zA-Z0-9]$").MatchString(value) {
+		errors = append(errors, fmt.Errorf("%s must contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long", k))
```

Comment:
Do we have a test that shows the API will accept a UUID here that starts with a number?  iirc the spec says the name must start with an alpha? if so, could you hand off the first regexp to `validation.isUUID` ?